db_stress_tool/db_stress_test_base.cc

```diff
@@ -693,9 +693,19 @@ void StressTest::OperateDb(ThreadState* thread) {
             snapshot, rand_column_family, column_family->GetName(),
             keystr,   status_at,          value_at,
             key_vec};
-        thread->snapshot_queue.emplace(
-            std::min(FLAGS_ops_per_thread - 1, i + FLAGS_snapshot_hold_ops),
-            snap_state);
+        auto hold_for = FLAGS_snapshot_hold_ops;
+        if (FLAGS_long_running_snapshots) {
+          // Hold 10% of snapshots for 10x more
+          if (thread->rand.OneIn(10)) {
```

Comment:
10% feels too large to me which gives too much share to long running snapshots. It feels that it should be something like one in 1000 or so.

Reply:
Currently we have 100M ops/thread in back box tests and take snapshot 1/10K ops. With 1000th of them being delayed we would end up with 10 delayed transactions. and there will be no space left for much delayed transactions.
Note that this is enabled randomly and not all the runs of the stress tests are exposed to so many long running snapshots.

Reply:
OK. I realized 10x and 100x are not very long, so it's fine.